src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -514,6 +514,7 @@ void StorageObjectStorage::drop()
         const auto [namespace_name, table_name] = DataLake::parseTableName(storage_id.getTableName());
         catalog->dropTable(namespace_name, table_name);
     }
+    configuration->drop(Context::getGlobalContextInstance());
```

Comment:
```suggestion
    /// We cannot use query context here, because drop is executed in the background.
    configuration->drop(Context::getGlobalContextInstance());
```